### ISSUE TITLE
Use namespace parameters or parameters mapping in pipeline call to prevent "Duplicate keys" error.

### ISSUE THREAD

**Issue by DOH-Manada** [kedro-org/kedro#929]
## Description
Is it possible to utilize the pipeline namespace in such a way that prevents Kedro from raising duplicate keys across parameters in different yaml files? I believe the current test combines all .yml files in the base/parameters folder and creates a dictionary out of them, in which the keys are the high-level parameter names without taking into consideration the YAML's name or the pipeline namespace.

## Context
This reduces the need for silly hard-code parameter naming across independent pipelines, e.g., column_date_pipeline1 & column_date_pipeline2 vs namespace1:column_date & namespace2:column_date.

This is my current implementation that will raise an error:
```yml
// base/parameters/pipeline1.yml

column_date : date # column for one dataset

// base/parameters/pipeline2.yml

column_date : date # column for a different dataset (or even the same dataset)
```

```python
// pipeline-registry.py

# Modular pipelines ===============================================================================
def create_pipeline1(inputs, outputs):
    instance = pipeline1.create_pipeline()
    pipe = Pipeline(
        [
            pipeline(
                instance,
                inputs={
                    "dataframe_input": inputs,
                    "column_date":"params:column_date",
                    },
                # perhaps parameters can solve my problem?
                outputs={"dataframe_output": outputs},
                namespace="do_something"
            ),
        ]
    )
    
    return pipe

def create_pipeline2(inputs, outputs):
    instance = pipeline1.create_pipeline()
    pipe = Pipeline(
        [
            pipeline(
                instance,
                inputs={
                    "dataframe_input": inputs,
                    "column_date":"params:column_date", # 
                    },
                # perhaps parameters can solve my problem?
                outputs={"dataframe_output": outputs},
                namespace="do_something_else"
            ),
        ]
    )
    
    return pipe
```

## Possible Implementation
I'm not sure now the `"params:"` key call occurs, but I'm sure something like `"params:{namespace}: value"`, `"params:{pipeline_name}: value"`, or simply `"{pipeline_name}: value"` is possible.

## Possible Alternatives
The alternative solution here is to use hard code names or use the parameters mapping dict in a clever way. I'm not sure how the parameters mapping in the pipeline call works to be honest.

**Comment by datajoely** [comment]
Hi @DOH-Manada  - we will be adding namespace to parameters to next major version of Kedro 0.18.0 since this will be a breaking change.

**Comment by MerelTheisenQB** [comment]
Closing this as support for this functionality will be released in Kedro `0.18.0`. The release should be coming in the next week(s).
